### ISSUE TITLE
Lists are not serialized correctly

### ISSUE THREAD

**Issue by olivierbellone** [stripe/stripe-java#229]
When [creating a product](https://stripe.com/docs/api/java#create_product) with the following code:

```java
Map<String, Object> productParams = new HashMap<String, Object>();
productParams.put("name", "T-shirt");
productParams.put("description", "Comfortable gray cotton t-shirts");
List<String> attributes = Arrays.asList("size", "gender");
productParams.put("attributes", attributes);
Product product = Product.create(productParams);
```

the request body will be encoded like this:
```
{
  description: "Comfortable gray cotton t-shirts",
  name: "T-shirt",
  attributes:
  {
    0: "size",
    1: "gender"
  }
}
```

Because the `attributes` list is not correctly serialized, the product creation then fails with:
```
error:
{
  type: "invalid_request_error",
  message: "Invalid array",
  param: "attributes"
}
```

`attributes` should be serialized as:
```
attributes: [
  "size",
  "gender"
]
```

**Comment by jloisel** [comment]
Happens on Order.all too, trying to pass ids to the request and it fails with invalid array.

**Comment by jloisel** [comment]
Pretty sure it affects all the requests which require to serialize lists.

**Comment by jloisel** [comment]
Can be reproduced using curl:

> curl -XGET https://api.stripe.com/v1/orders?ids%5B0%5D=<OrderId>&status=paid -u <Test Key>

**Comment by brandur** [comment]
Wow, was just looking into this a bit and it seems that [this was done on purpose](https://github.com/stripe/stripe-java/blob/brandur-pagination/src/test/java/com/stripe/net/LiveStripeResponseGetterTest.java#L55,L66). As you can see in that test, it knowingly changes a list into an object instead of projecting it into an array.

I'm not sure why (maybe the server is supposed to support this?), but will try to work my way down the list of top contributors to ask when I get to the office.

/cc @kyleconroy In case he has any additional context.

**Comment by kyleconroy** [comment]
@brandur Indeed. This is how `/v1/accounts` works. We should check in on the other bindings to make sure that order creation is working properly in the other bindings. We should probably put this special case in the Account class and make the general case do the correct thing.

**Comment by brandur** [comment]
@kyleconroy Thanks! Do you think that I might just be able to apply a universal fix for this to Java package? I took a look at the API docs for `/v1/accounts`, and as far as I can tell, all the mutable collection fields seem to be able to arrays. Is it possible that the necessity for the `0 => ..., 1 => ..., 2 => ...` style syntax was phased out?

**Comment by brandur** [comment]
This still needs a release, but was fixed by #231.
